Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -5344,7 +5344,7 @@ bool RenderBox::hasUnsplittableScrollingOverflow() const
 
 bool RenderBox::isUnsplittableForPagination() const
 {
-    return isReplacedOrAtomicInline()
+    return shouldTreatChildAsReplaced()
```

Comment:
This should just be along the lines of isReplacedOrAtomicInline() || isFormControlElement() instead of a mysterious shouldTreatChildAsReplaced() function...and we should get rid of shouldTreatChildAsReplaced.

Reply:
actually let's do that in a follow-up PR.